apm-agent-plugins/apm-spring-scheduled-plugin/src/main/java/co/elastic/apm/agent/spring/scheduled/SpringScheduledTransactionNameInstrumentation.java

```diff
@@ -0,0 +1,104 @@
+import co.elastic.apm.agent.impl.stacktrace.StacktraceConfiguration;
+import co.elastic.apm.agent.impl.transaction.TraceContext;
+import co.elastic.apm.agent.impl.transaction.TraceContextHolder;
+import co.elastic.apm.agent.impl.transaction.Transaction;
+import net.bytebuddy.asm.Advice;
+import net.bytebuddy.description.NamedElement;
+import net.bytebuddy.description.method.MethodDescription;
+import net.bytebuddy.description.type.TypeDescription;
+import net.bytebuddy.matcher.ElementMatcher;
+import net.bytebuddy.matcher.ElementMatchers;
+
+import static co.elastic.apm.agent.bci.bytebuddy.CustomElementMatchers.isInAnyPackage;
+import static net.bytebuddy.matcher.ElementMatchers.declaresMethod;
+import static net.bytebuddy.matcher.ElementMatchers.isAnnotatedWith;
+import static net.bytebuddy.matcher.ElementMatchers.named;
+
+public class SpringScheduledTransactionNameInstrumentation extends ElasticApmInstrumentation {
+
+    @VisibleForAdvice
+    public static final Logger logger = LoggerFactory.getLogger(SpringScheduledTransactionNameInstrumentation.class);
+
+    private Collection<String> applicationPackages = Collections.emptyList();
+
+    @Advice.OnMethodEnter(suppress = Throwable.class)
+    private static void setTransactionName(@SimpleMethodSignature String signature, @Advice.Origin Class<?> clazz, @Advice.Local("transaction") Transaction transaction) {
+        if (tracer != null) {
+            TraceContextHolder<?> active = tracer.getActive();
+            if (active == null) {
+                transaction = tracer.startTransaction(TraceContext.asRoot(), null, clazz.getClassLoader())
+                        .withName(signature)
+                        .withType("scheduled")
+                        .activate();
+
+            } else {
+                logger.debug("Not creating transaction for method {} because there is already a transaction running ({})", signature, active);
+            }
+        }
+    }
+
+    @Advice.OnMethodExit(suppress = Throwable.class, onThrowable = Throwable.class)
+    public static void onMethodExit(@Nullable @Advice.Local("transaction") Transaction transaction,
+                                    @Advice.Thrown Throwable t) {
+        if (transaction != null) {
+            transaction.captureException(t)
+                    .deactivate()
+                    .end();
+        }
+    }
+
+    @Override
+    public void init(ElasticApmTracer tracer) {
+        applicationPackages = tracer.getConfig(StacktraceConfiguration.class).getApplicationPackages();
+    }
+
+    @Override
+    public ElementMatcher<? super TypeDescription> getTypeMatcher() {
+        return isInAnyPackage(applicationPackages, ElementMatchers.<NamedElement>none())
+                .and(declaresMethod(getMethodMatcher()));
+    }
+
+    @Override
+    public ElementMatcher<? super MethodDescription> getMethodMatcher() {
+        return isAnnotatedWith(named("org.springframework.scheduling.annotation.Scheduled"));
```

Comment:
You may also want to add these annotations:
`org.springframework.scheduling.annotation.Schedules`, `javax.ejb.Schedule`, `javax.ejb.Schedules`. Maybe it makes sense to rename the plugin to `apm-scheduled-annotation-plugin` instead?

Reply:
Added the annotations and renamed the plugin